src/bc_protocol.rs

```diff
@@ -266,20 +271,48 @@ impl BcCamera {
 
         sub_video.send(start_video)?;
 
+        let mut media_sub = MediaDataSubscriber::from_bc_sub(&sub_video);
+
         loop {
-            trace!("Getting video message...");
-            let msg = sub_video.rx.recv_timeout(RX_TIMEOUT)?;
-            if let BcBody::ModernMsg(ModernMsg {
-                binary: Some(binary),
-                ..
-            }) = msg.body
-            {
-                trace!("Got {} bytes of video data", binary.len());
-                data_out.write_all(binary.as_slice())?;
-            } else {
-                warn!("Ignoring weird video message");
-                debug!("Contents: {:?}", msg);
-            }
+            let binary_data = media_sub.get_media_packet_of_kind(
+                &vec![
+                    MediaDataKind::VideoDataIframe,
+                    MediaDataKind::VideoDataPframe,
+                ],
+                RX_TIMEOUT,
+            )?;
+            // We now have a complete interesting packet. Send it to gst.
+            // Process the packet
+            match binary_data.kind() {
+                MediaDataKind::VideoDataIframe | MediaDataKind::VideoDataPframe => {
+                    data_out.write_all(binary_data.body())?;
+                }
+                MediaDataKind::AudioDataAac | MediaDataKind::AudioDataAdpcm => {
+                    // We should not have a packet of this kind as we skip them until we get
+                    // a packet of interest
+                    unreachable!();
+                }
+                MediaDataKind::InfoData => {
+                    // We should not have a packet of this kind as we skip them until we get
+                    // a packet of interest
+                    unreachable!();
```

Comment:
I have these as unreachable at the moment mostly to ensure that I don't get unexpected data sneaking though during testing. When we've stress tested it enough I'd like to remove them.

Reply:
Ok I have removed this and some other superfluous codes